src/sync.c

```diff
@@ -180,3 +179,27 @@ mpd_sync_recv_line(struct mpd_async *async, const struct timeval *tv0)
 			return NULL;
 	}
 }
+
+struct mpd_binary
+mpd_sync_recv_binary(struct mpd_async *async, const struct timeval *tv0, const unsigned binary)
+{
+	struct timeval tv, *tvp;
+	struct mpd_binary data;
+	data.size = 0;
+	data.data = NULL;
+
+	if (tv0 != NULL) {
+		tv = *tv0;
+		tvp = &tv;
+	} else
+		tvp = NULL;
+
+	while (true) {
+		data = mpd_async_recv_binary(async, binary);
+		if (data.data != NULL)
```

Comment:
Why does this check the `data` pointer instead of the `size`? Unfortunately, this function is undocumented, which means I can't verify if your check is correct, but I guess this check is wrong, and only works due to an undocumented implementation detail of `mpd_async_recv_binary()`.

Reply:
changed